devtools/gradle/src/main/java/io/quarkus/gradle/AppModelGradleResolver.java

```diff
@@ -192,6 +194,10 @@ private ModuleIdentifier getModuleId(ResolvedArtifact a) {
     private AppDependency toAppDependency(ResolvedArtifact a) {
         final String[] split = a.getModuleVersion().toString().split(":");
         final AppArtifact appArtifact = new AppArtifact(split[0], split[1], split[2]);
+        final ComponentIdentifier id = a.getId().getComponentIdentifier();
+        if (id instanceof ProjectComponentIdentifier) {
+            appArtifact.setSubModule(true);
+        }
```

Comment:
You can also replace this snippet with `appArtifact.setSubModule(id instanceof ProjectComponentIdentifier)` :wink:

Reply:
Or simply `appArtifact.setSubModule(a.getId().getComponentIdentifier() instanceof ProjectComponentIdentifier);`